### DISCUSSION TITLE
Add OneDrive/Yandex.Disk/Mail.ru virtual folders support

### DISCUSSION THREAD

**Discussion by nkzmn** [doublecmd/doublecmd#1296] [Suggestions]
Just like `This PC` virtual folder is supported on recent Windows, it would be nice to have also support for other virtual folders, including `OneDrive`/`Yandex.Disk`/`Mail.ru`.

**Comment by Skif-off** [comment]
Multiple operating systems and multiple protocols? Could this be real?

**Comment by alexx2000** [comment]
`OneDrive` is not a virtual folder, with installed official client it is available as normal folder. Also you can use [Cloud](https://www.ghisler.com/plugins.htm#filesys) plugin which supports many cloud services and [CloudMailRu](https://github.com/pozitronik/CloudMailRu/releases/tag/v300722) plugin for `Mail.ru`.

@Skif-off, technically it is possible. But it is hard. Cloud services does not use standard protocols so they can change protocol at any time and broke program. Standalone plugin looks better in this case.

**Reply by nkzmn** [reply]
Custom cloud protocol implementations is a separate complicated matter and indeed it is better to have them as external plugins.

Actually I asked about another, more simple (as I feel) thing.

When I open File Explorer on Windows I see the following categories in its left panel: `Quick Access`, `OneDrive` (its client is installed by default), `Yandex.Disk` (my system has this client installed), `This PC`. I would like to see all these categories (in sync with File Explorer) in Double Commander too (currently it shows only `This PC`) . Is it possible?

**Reply by alexx2000** [reply]
It is a place for special, virtual folders which is not accessible directly. These cloud folders are available in the user's directory. You can open it using `~` button.

**Comment by RoyiAvital** [comment]
@alexx2000 , Does the _Total Commander_ work on macOS / Linux?
I'd assume they work on Windows only.

**Reply by alexx2000** [reply]
No. But for Linux (and I guess macOS) exists alternative [cloud_storage ](https://github.com/doublecmd/doublecmd/wiki/Plugins#user-content-cloud_storage) plugin.

**Reply by rich2014** [reply]
@RoyiAvital 

on macOS, it's [Offical cloud plugin for macOS 2025](https://github.com/doublecmd/doublecmd/discussions/2300)
